Week5

nois0720 opened the pull request:
책 과제 수행
개인 과제 첫 커밋. API서버와 통신하여 로그인, 회원가입 기능 포함.

nois0720 committed:
add book assn

nois0720 committed:
add spec login, signup

nois0720 committed:
add imagetable view and upload view

wanbok commented on week5/Photorama/Photorama/PhotoStore.swift:
```diff
@@ -0,0 +1,89 @@
+import UIKit
+
+enum ImageResult {
+    case Success(UIImage)
```
Swift 3부터 enum의 case들이 소문자로 시작되도록 만드는걸 권장하고 있습니다.

nois0720 replied:
네 알겠습니다. 개인과제에도 enum 많이 쓰고잇는데 적용하겠습니다!

wanbok commented on week5/Photorama/Photorama/FlickrAPI.swift:
```diff
@@ -0,0 +1,106 @@
+    }
+    
+    private static func photoFromJSONObject(json: [String : AnyObject]) -> Photo? {
+        guard let photoID = json["id"] as? String,
```
👍

wanbok commented on week5/Photorama/Photorama/PhotoStore.swift:
```diff
@@ -0,0 +1,89 @@
+        let task = session.dataTask(with: request) {
+            (data, response, error) -> Void in
+            
+            /* 19장 동메달 과제. (응답 정보 출력하기) */
```
👍

wanbok commented on week5/Photorama/Photorama/PhotoStore.swift:
```diff
@@ -0,0 +1,89 @@
+    func processImageRequest(data: Data?, error: Error?) -> ImageResult {
+        guard let imageData = data,
+            let image = UIImage(data: imageData) else {
+                if data == nil {
```
3항 연산을 쓸 수도 있는 부분으로 보여집니다만, 안해도 상관없긴 합니다.
.Failure(data == nil ? error! : PhotoError.ImageCreationError)

wanbok commented on week5/Photorama/Photorama/PhotosViewController.swift:
```diff
@@ -0,0 +1,76 @@
+    }
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        guard segue.identifier == "ShowPhoto",
```
👍

wanbok reviewed (commented):
지난번에 언급된 지적사항들이 충실히 반영되어있네요. 크게 지적할 사항이 없어보입니다. 👍

nois0720 committed:
- 1차 리팩토링 완료(AuthenticationCenter.swift, PhotoStore 제거. 제거한 파일의 API 관련 기능 ImageBoardAPI extension에서 구현)
- detail view/controller 추가
- Model 추가 및 수정(Photo수정, PhotoForUpload추가)

nois0720 committed:
삭제 수정 기능 추가.
테이블 뷰 새로고침 추가.
스토리보드를 login, main으로 나눔
2차 리팩토링(enum 목록 lower-camel-case로 변경, API 함수 명 변경)

nois0720 committed:
modify enum uppercase error
